modules/linear_boltzmann_solvers/lbs_solver/io/flux_moments_io.cc

```diff
@@ -15,99 +16,45 @@ LBSSolverIO::WriteFluxMoments(
-                            "  uint64_t    cell_global_id\n"
-                            "  uint64_t    num_cell_nodes\n"
-                            "  Each node:\n"
-                            "    double   x_position\n"
-                            "    double   y_position\n"
-                            "    double   z_position\n"
-                            "Each record:\n"
-                            "  uint64_t    cell_global_id\n"
-                            "  uint64_t    node\n"
-                            "  uint64_t    moment\n"
-                            "  uint64_t    group\n"
-                            "  double      value\n";
-
-  int header_size = (int)header_info.length();
-  char header_bytes[num_bytes];
-  memset(header_bytes, '-', num_bytes);
-  strncpy(header_bytes, header_info.c_str(), std::min(header_size, num_bytes - 1));
-  header_bytes[num_bytes - 1] = '\0';
-
-  file << header_bytes;
-
-  // Write macro data
+  const auto& grid = lbs_solver.Grid();
+  const auto& discretization = lbs_solver.SpatialDiscretization();
+
   const auto& uk_man = lbs_solver.UnknownManager();
   const auto NODES_ONLY = UnknownManager::GetUnitaryUnknownManager();
-  auto& discretization = lbs_solver.SpatialDiscretization();
-  auto& grid = lbs_solver.Grid();
-  const uint64_t num_local_cells = grid.local_cells.size();
-  const uint64_t num_local_nodes = discretization.GetNumLocalDOFs(NODES_ONLY);
-  const uint64_t num_moments = lbs_solver.NumMoments();
-  const uint64_t num_groups = lbs_solver.NumGroups();
-  const auto num_local_dofs = discretization.GetNumLocalDOFs(uk_man);
-  OpenSnLogicalErrorIf(src.size() != num_local_dofs, "Incompatible flux moments vector provided..");
 
-  file.write((char*)&num_local_cells, sizeof(uint64_t));
-  file.write((char*)&num_local_nodes, sizeof(uint64_t));
-  file.write((char*)&num_moments, sizeof(uint64_t));
-  file.write((char*)&num_groups, sizeof(uint64_t));
+  uint64_t num_local_cells = grid.local_cells.size();
+  uint64_t num_local_nodes = discretization.GetNumLocalDOFs(NODES_ONLY);
+  uint64_t num_moments = lbs_solver.NumMoments();
+  uint64_t num_groups = lbs_solver.NumGroups();
+  uint64_t num_local_dofs = num_local_nodes * num_moments * num_groups;
+  OpenSnLogicalErrorIf(src.size() != num_local_dofs, "Incompatible flux moments vector provided.");
 
-  // Write nodal positions
-  for (const auto& cell : grid.local_cells)
-  {
-    const uint64_t cell_global_id = cell.global_id;
-    const uint64_t num_cell_nodes = discretization.GetCellNumNodes(cell);
-
-    file.write((char*)&cell_global_id, sizeof(uint64_t));
-    file.write((char*)&num_cell_nodes, sizeof(uint64_t));
-
-    const auto nodes = discretization.GetCellNodeLocations(cell);
-    for (const auto& node : nodes)
-    {
-      file.write((char*)&node.x, sizeof(double));
-      file.write((char*)&node.y, sizeof(double));
-      file.write((char*)&node.z, sizeof(double));
-    } // for node
-  }   // for cell
```

Comment:
You don't want to remove this. The purpose of this code is:  You obtain a flux vector on mesh1. You try use the stored vector with mesh2. We need to check that the meshes are the same. Just the number of cells and nodes is not unique enough to prevent users to use the stored vector with a different mesh. This is why we store the vertex location and in read they are checked that they map on the mesh we already loaded.

Reply:
Also, the global IDs and number of nodes per cell needs to be stored in the file. In general case, every cell could have different number of nodes (the code supports polygon/polyhedron elements). And the global IDs are used in parallel, so that each rank picks its cells. It is this bit in the read function:
```
if (grid.IsCellLocal(cell_global_id))
```
So, bring these bits back.

Reply:
A few comments/questions on this for @andrsd. First, the angular flux files in #381 (and preceding code) do not have grid information. Should this be added there? Also, would it be better to introduce a separate h5 file for the grid for these purposes? When we have transients, for example, if we dump the flux moments per time step we would then be writing the same grid information over and over. 

@quincy-huhn98 I would make two groups in the h5 file, one called "grid" or "mesh" and the other called "flux_moments" to separate the information. Everything currently written should go in the "flux_moments" group.

Reply:
I would start with a single file for both steady state and transient. Everything is going to be much simpler - both code-wise and easier for users (in case they need to transfer those files, post-processing, etc.).

Also, if the angular fluxes could be stored in the same file as scalar fluxes, even better :-) But I understand those get real huge real fast. 

One big reason I can see to start doing separate files is when the file size gets so big that the file system cannot handle that - if that's the direction of your thinking...

Reply:
I was still working out how to implement the grid, so I appreciate the direction @andrsd